eng/common/TestResources/New-TestResources.ps1

```diff
@@ -194,14 +198,18 @@ $serviceName = if (Split-Path -IsAbsolute  $ServiceDirectory) {
     $ServiceDirectory
 }
 
-# Format the resource group name based on resource group naming recommendations and limitations.
-$resourceGroupName = if ($CI) {
-    $BaseName = 't' + (New-Guid).ToString('n').Substring(0, 16)
-    Write-Verbose "Generated base name '$BaseName' for CI build"
+if ($CI) { 
+  $BaseName = 't' + (New-Guid).ToString('n').Substring(0, 16)
+  Write-Verbose "Generated base name '$BaseName' for CI build"
+}
 
-    "rg-{0}-$BaseName" -f ($serviceName -replace '[\\\/:]', '-').Substring(0, [Math]::Min($serviceName.Length, 90 - $BaseName.Length - 4)).Trim('-')
+$rgName = if ($ResourceGroupName -ne '') {
```

Comment:
Tip 2: you could also just use `$ResourceGroupName` as the property - it's not passed by ref or anything, so you can use variables defined as parameters as locals as well.